is_core/forms/models.py

```diff
@@ -221,20 +222,22 @@ class SmartModelForm(six.with_metaclass(SmartModelFormMetaclass, SmartFormMixin,
 
     def __init__(self, *args, **kwargs):
         # Set values must be ommited
-        readonly_exclude = kwargs.get('initials', {}).keys()
+        readonly_exclude = kwargs.get('initials',
+                                      {}).keys()
 
         super(SmartModelForm, self).__init__(*args, **kwargs)
 
         opts = self._meta
         self.humanized_data = humanized_model_to_dict(self.instance, self.base_readonly_fields, opts.fields,
-                                                      itertools.chain(opts.exclude or (), readonly_exclude or  ()))
+                                                      itertools.chain(opts.exclude or (), readonly_exclude or ()))
 
         if self.instance.pk:
             for rel_object in opts.model._meta.related_objects:
                 if rel_object.name in self.fields and rel_object.name not in self.initial:
-                    self.initial[rel_object.name] = list(
-                        getattr(self.instance, rel_object.name).values_list('pk', flat=True)
-                    )
+                    if hasattr(self.instance, rel_object.name):
```

Comment:
I think the bug is still there, but a solution should be different. DemandForm has an attribute `communication`, but `demand` field on Communication has no `related_name` (it's set to default `communication_set`). Line 237 uses `rel_object.name` for a field name on a model. And there is the bug. `rel_object.get_accessor_name()` should be used, instead of `rel_object.name`. Because `rel_object.name` returns a class name and `rel_object.get_accessor_name()` returns a real field name. Does it make sense?